src/main.js

```diff
@@ -20,46 +25,63 @@ class Main {
     }
 
     init() {
+        this.codeEditor = new CodeEditor(this.textArea);
+
+        this.setSupportedLangs();
+        this.languageSelect.value = config.DEFAULT_LANG;
+        this.codeEditor.setLanguage(config.DEFAULT_LANG);
+
         // Check to seed if there is a query param for the snippet
-        let snippetID = this.getQueryParam(config.SNIPPET_QUERY_PARAM);
-        if (snippetID != "") {
+        let snippetID = Utils.getQueryParam(config.SNIPPET_QUERY_PARAM);
+        if (snippetID === "") {
+            this.codeEditor.setContent(config.DEFAULT_CONTENT, false)
+        } else {
             // We have an ID of a snippet
             let self = this;
-            CodeEditor.xmlReq("/snippet/" + snippetID, "GET")
+            Utils.xmlReq("/snippet/" + snippetID, "GET")
                 .then(function(resp) {
-                    self.textArea.value = resp.snippet;
                     self.languageSelect.value = resp.language;
-                    self.codeEditor = new CodeEditor(self.textArea);
+                    self.codeEditor.setContent(resp.snippet);
                     self.codeEditor.setLanguage(resp.language);
                 }).catch(function(err) {
                     console.error("Error: ", err);
+                    self.codeEditor.setContent(config.SNIPPET_NOT_FOUND, false);
                 });
-        } else {
-            this.codeEditor = new CodeEditor(this.textArea);
         }
+        // Ad event handler for editor being focussed
+        this.codeEditor.on('editor-focus', this.editorFocused.bind(this));
+    }
+
+    editorFocused() {
+        if (!this.codeEditor.shouldPersist()) {
+            this.codeEditor.setContent("");
+        }
+    }
 
+    setSupportedLangs() {
+        for (let lang of this.codeEditor.getLanguages()) {
+            let option = document.createElement("option");
+            option.value = lang.name;
+            option.innerHTML = lang.name;
+            this.languageSelect.appendChild(option);
+        }
     }
 
     shareClicked(event) {
         let self = this;
         this.codeEditor.saveAndGetLink()
             .then(function(link) {
+              if (link != null) {
```

Comment:
I wrapped in a non strict null (to cover undefined and null). 
If clicked share again without changing code, because we resolve without returning a link it was filling the link box with an undefined